python/lsst/source/injection/utils/consolidate_injected_deepCoadd_catalogs.py

```diff
@@ -0,0 +1,609 @@
+
+    Parameters
+    ----------
+    catalog_dict: `dict`
+        A dictionary with photometric bands for keys and astropy tables for
+        items.
+    skymap: `lsst.skymap.BaseSkyMap`
+        A base skymap.
+    tract: `int`
+        The tract where sources have been injected.
+    pixel_match_radius: `float`
+        Match radius in pixels to use for self-matching catalogs across
+        different bands.
+    get_catalogs_from_butler: `bool`
+        Optional parameter to specify whether or not the input catalogs are
+        loaded with a Butler.
+    col_ra: `str`
+        Column name for right ascension (in degrees).
+    col_dec: `str`
+        Column name for declination (in degrees).
+    col_mag: `str`
+        Column name for magnitude.
+    isPatchInnerKey: `str`
+        Column name for the isPatchInner flag.
+    isTractInnerKey: `str`
+        Column name for the isTractInner flag.
+    isPrimaryKey: `str`
+        Column name for the isPrimary flag.
+    injectionKey: `str`
+        Column name for the injection flag.
+
+    Returns
+    -------
+    multiband_catalog: `astropy.table.Table`
+        A single table containing all information of the separate
+        tables in catalog_dict
+    """
+    tractInfo = skymap.generateTract(tract)
+    # If patch numbers are not loaded via dataIds from the butler, manually
+    # load patch numbers from source positions.
+    if not get_catalogs_from_butler:
+        _get_patches(catalog_dict, tractInfo)
+
+    # Convert the pixel match radius to degrees.
+    tractWcs = tractInfo.getWcs()
+    pixel_scale = tractWcs.getPixelScale()
+    match_radius = pixel_match_radius * pixel_scale.asDegrees()
+    bands = list(catalog_dict.keys())
+    if len(bands) > 1:
+        # Match the catalogs across bands.
+        output_catalog = _make_multiband_catalog(bands, catalog_dict, match_radius)
+    else:
+        output_catalog = catalog_dict[bands[0]]
+        output_catalog.rename_column(col_mag, f"{bands[0]}_{col_mag}")
+    # Remove sources outside tract boundaries.
+    out_of_tract_bounds = []
+    for index, (ra, dec) in enumerate(list(zip(output_catalog[col_ra], output_catalog[col_dec]))):
+        point = SpherePoint(ra * degrees, dec * degrees)
+        if not tractInfo.contains(point):
+            out_of_tract_bounds.append(index)
+    output_catalog.remove_rows(out_of_tract_bounds)
+    # Assign flags.
+    patches = list(set(output_catalog["patch"]))
+    setPrimaryFlags(
+        catalog=output_catalog,
+        skyMap=skymap,
+        tractInfo=tractInfo,
+        patches=patches,
+        col_ra=col_ra,
+        col_dec=col_dec,
+        isPatchInnerKey=isPatchInnerKey,
+        isTractInnerKey=isTractInnerKey,
+        isPrimaryKey=isPrimaryKey,
+        injectionKey=injectionKey,
+    )
+    # Replace old injection_id column with a new injected_id column.
+    output_catalog["injection_id"] = list(range(len(output_catalog)))
```

Comment:
Is this what we want to do here? Losing our injection IDs makes it harder to circle back to an input, if required?